### PR TITLE
Co 3084     Migrer account_reconcile_compassion

### PR DESCRIPTION
**Pull request by SylvainLosey**
Test fail because module depends on [account_bank_statement_import_camt_details](https://github.com/OCA/bank-statement-import/pull/249) OCA migration

### CODE REVIEW

File: account_reconcile_compassion/models/res_config.py [hunk 1]
```diff
@@ -19,28 +18,28 @@ class AccountConfigSettings(models.TransientModel):
     currency exchange.
     """
 
-    _inherit = 'account.config.settings'
+    _inherit = "account.config.settings"
```
Comment (josueBulle) [inline account_reconcile_compassion/models/res_config.py]: 'account.config.settings' should not exist.
This class should _inherit 'res.config.settings' and override the `get_values` and `set_values` methods.
Views need adaptations

File: account_reconcile_compassion/models/res_config.py [hunk 2]
```diff
@@ -19,28 +18,28 @@ class AccountConfigSettings(models.TransientModel):
...
 
     currency_exchange_analytic_account = fields.Many2one(
-        'account.analytic.account',
-        default=lambda s: s.get_currency_exchange_analytic_account()
+        "account.analytic.account",
+        default=lambda s: s.get_currency_exchange_analytic_account(),
     )
 
     @api.multi
     def set_currency_exchange_analytic_account(self):
```
Comment (josueBulle) [inline account_reconcile_compassion/models/res_config.py]: The `set_currency...` and `get_currency...` methods can be deleted.
Calls to `get_currency_` will then need refactoring

**Review by josueBulle — CHANGES_REQUESTED** [review]

**Comment by josueBulle** [comment]
Please check your PR on the OCA too, tests are failing.

**Comment by ecino** [comment]
Please make PR of OCA on the fork from CompassionCH. Then update oca_dependencies.txt to fetch the CompassionCH fork so that the requirement can be found.

File: oca_dependencies.txt [hunk 3]
```diff
@@ -32,6 +32,7 @@ web
 website https://github.com/CompassionCH/website 11.0
 website-cms
 l10n-switzerland https://github.com/CompassionCH/l10n-switzerland 11.0-mig-l10n_ch_pain_direct_debit
+bank-statement-import https://github.com/CompassionCH/bank-statement-import 11.0
```
Comment (ecino) [inline oca_dependencies.txt]: Sorry, I changed the branch name to make it more clear.
```suggestion
bank-statement-import https://github.com/CompassionCH/bank-statement-import 11.0-camt-details
```

**Review by ecino — CHANGES_REQUESTED** [review]

**Comment by ecino** [comment]
Please correct as the test `File "/home/travis/build/CompassionCH/compassion-switzerland/account_reconcile_compassion/tests/test_bank_account_assignation.py", line 26, in setUp
`     "journal_id": self.env.ref("l10n_ch_lsv_dd.lsv_account_journal").id,`

The module lsv_dd doesn't exist anymore so you cannot reference this anymore. You should use another demo journal instead.

**Comment by ecino** [comment]
Please rebase onto 11.0 to fix the conflicts

**Comment by ecino** [comment]
Tests are still failing.

**Comment by SylvainLosey** [comment]
```
# test get_move_lines_for_reconciliation method
self.assertEqual(len(bank_statement_line.get_move_lines_for_reconciliation()), 12)
```

This was previously asserting that it was equal to 0, which by reading the related functions I don't understand.

**Pull request closed without merging**